python/graphs/kuhn.py

```diff
@@ -0,0 +1,27 @@
+
+def dfs(adj, visited, right, left, u):
+    visited[u] = True
+    for v in adj[u]:
+        # If the adj node unpaired then pair it to this.
+        # Otherwise try and pair the adj one's pair to somewhere else.
+        if (left[v] == -1 or
+            (not visited[left[v]] and
+             dfs(adj, visited, right, left, left[v]))):
+            left[v] = u
+            right[u] = v
+            return True
+    return False
+
+
```

Comment:
it'd be good to get an example of the inputs or something, for quick reference, instead of trying to remember/derive what type `adj` is supposed to be.

Reply:
Also, mentioning here the time complexity would be useful.